tests/test_estimate_params.py

```diff
@@ -0,0 +1,965 @@
+        self.assertEqual(
+            'Times, inputs, and outputs must be same length for the run at index 0. Length of times: 9, Length of inputs: 8, Length of outputs: 8',
+            str(cm.exception)
+        )
+
+        # Without wrapper
+        with self.assertRaises(ValueError) as cm:
+            m.estimate_params(times=results.times, inputs=wrongInputLen, outputs= wrongOutputs)
+        self.assertEqual(
+            'Times, inputs, and outputs must be same length for the run at index 0. Length of times: 9, Length of inputs: 8, Length of outputs: 8',
+            str(cm.exception)
+        )
+
+        # Length error expected, 1, 9, 1.
+        with self.assertRaises(ValueError) as cm:
+            m.estimate_params(times=[[results.times]], inputs=[results.inputs], outputs=[[results.outputs]])
+        self.assertEqual(
+            'Times, inputs, and outputs must be same length for the run at index 0. Length of times: 1, Length of inputs: 9, Length of outputs: 1',
+            str(cm.exception)
+        )
+
+
+        # Passing in incorrect Runs
+        with self.assertRaises(ValueError):
+            m.estimate_params(wrongData)
+
+        m.parameters['thrower_height'] = 1.5
+        m.parameters['throwing_speed'] = 25
+
+        # Testing functionality works without having a parent wrapper
+        m.estimate_params(times=results.times, inputs=results.inputs, outputs=results.outputs)
+        for key in keys:
+            self.assertAlmostEqual(m.parameters[key], gt[key], 2)
+
+        # Differnet types of wrappers should not affect function
+        # Testing functionality works with having only a few parameters defined in wrapper sequences
+        m.parameters['thrower_height'] = 1.5
+        m.parameters['throwing_speed'] = 25
+
+        m.estimate_params(times=results.times, inputs=(results.inputs), outputs=(results.outputs))
+        for key in keys:
+            self.assertAlmostEqual(m.parameters[key], gt[key], 2)
+        
+        # Same as last test, but inputs is has tuple wrapper sequence insteads.
+        m.parameters['thrower_height'] = 1.5
+        m.parameters['throwing_speed'] = 25
+
+        m.estimate_params(times=results.times, inputs=(results.inputs), outputs=[results.outputs])
+        for key in keys:
+            self.assertAlmostEqual(m.parameters[key], gt[key], 2)
+
+        # Cannot pass in Sets
+        with self.assertRaises(TypeError):
+            m.estimate_params(times=set(results.times), inputs=results.inputs, outputs=[results.outputs])
+
+        # Missing inputs.
+        with self.assertRaises(ValueError) as cm:
+            m.estimate_params(times=[results.times], outputs=[results.outputs])
+        self.assertEqual(
+            'Missing keyword arguments inputs',
+            str(cm.exception)
+        )
+
+        #  'input' is not a parameter, so techincally not defining the parameter inputs.
```

Comment:
```suggestion
        #  'input' is not a keyword argument, so technically not defining the inputs.
```
Same issue with the term "parameter"